src/matrix/push/Pusher.ts

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type {HomeServerApi} from "../net/HomeServerApi.js";
+import type {ILogItem} from "../../logging/types";
+
+export interface IPusherDescription {
+    kind: "http" | "email" | "null";
+    lang: string;
+    device_display_name: string;
+    app_display_name: string;
+    app_id: string;
+    pushkey: string;
+    data: IPusherData;
+    append?: boolean;
+    profile_tag?: string;
+}
+
+interface IPusherData {
+    format?: string;
+    url?: string;
+    // todo: where did this come from?
```

Comment:
This is the push data that the homeserver will pass to the notification server, sygnal, and is defined here for web push: https://github.com/matrix-org/sygnal/blob/main/docs/applications.md#push-key-and-expected-push-data

Reply:
Also see where the enablePush method in NotificationService.

Reply:
Thank you! Have made slight changes.